docs/design/keyboard-shortcuts.md

```diff
@@ -256,6 +256,69 @@ When using custom keyboard shortcuts on the web, some keyboard shortcuts that ar
 - Ctrl+W
 - Ctrl+PgUp/PgDn
 
+## Enable end users to customize the keyboard shortcuts for your add-in (preview)
+
+Your add-in can enable users to reassign the actions of the add-in to alternate keyboard combinations.
+
+> [!IMPORTANT]
+> The features described in this section are currently in preview and subject to change. They are not currently supported for use in production environments. To try the preview features, you will need to [join Office Insider](https://insider.office.com/join).
+> A good way to try out preview features is by using a Microsoft 365 subscription. If you don't already have a Microsoft 365 subscription, you can get one by joining the [Microsoft 365 developer program](https://developer.microsoft.com/office/dev-program).
+
+> [!NOTE]
+> The APIs described in this section require the [KeyboardShortcuts 1.1](../reference/requirement-sets/keyboard-shortcuts-requirement-sets.md) requirement set.
+
+To enable a user to customize the keyboard shortcuts, your add-in will need to provide a UI in which the user can assign new key combinations to the add-in's actions. The UI should present the user with the available actions and, for each, provide an input method, such as a text box, for the user to enter a key combination string. The UI should validate that the strings are the proper form. For example, there can be no spaces and keys must be separated by the "+" symbol. If the user is logged into Office, the custom combinations are saved in the user's roaming settings. If the user is not logged in, the customizations last only for the current session of the add-in.
+
+Use the [Office.actions.replaceShortcuts](/javascript/api/office/office.actions#replaceShortcuts) method to assign a user's custom keyboard combinations to your add-ins actions. The method takes a parameter of type `{[actionId:string]: string}`, where the `actionId`s are a subset of the action IDs that are defined in the add-in's extended manifest JSON. The values are the user's preferred key combinations. The following is an example in which some hypothetical custom key combinations are hardcoded. In an actual add-in, you create a function that gets them from the UI.
```

Comment:
We should also remove the 'In an actual add-in, you create a function that gets them from the UI' because there could be several sources that these are fetched from in a real add-in scenario.